rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMStateCaching.java

```diff
@@ -121,51 +94,23 @@ private static void init_() {
         }
     }
 
-    private static void run_() {
-        rmUpdater = new Thread(new Runnable() {
-            @Override
-            public void run() {
-                boolean isDebugEnabled = logger.isDebugEnabled();
-                long startTime = 0;
-
-                while (!kill) {
-                    try {
-                        if (isDebugEnabled) {
-                            startTime = System.currentTimeMillis();
-                        }
-
-                        state = PAFuture.getFutureValue(rm.getRMInitialState());
-
-                        if (isDebugEnabled) {
-                            logger.debug("Updated RM initial state in " + (System.currentTimeMillis() - startTime) +
-                                         "ms");
-                        }
-                    } catch (Throwable t) {
-                        logger.error("Exception occurrend while updating RM state cache, connection reset", t);
-                        init_();
-                    }
-
-                    new Sleeper(refreshInterval, logger).sleep();
-                }
-            }
-        });
-        rmUpdater.setName("RM Initial State Cache Updater");
-        rmUpdater.setDaemon(true);
-        rmUpdater.start();
-    }
-
     /**
-     * @return cached RM State as returned by {@link RMProxyUserInterface#getMonitoring()}
+     * @return cached RM State as returned by {@link RMProxyUserInterface#getRMInitialState(long)}
      */
-    public static RMInitialState getRMInitialState() {
-        return state;
-    }
+    public static RMInitialState getRMInitialState(long counter) {
+        try {
+            long startTime = System.currentTimeMillis();
 
-    /**
-     * Stop the RM State polling thread.
-     */
-    public static void kill() {
-        RMStateCaching.kill = true;
-    }
+            final RMInitialState state = PAFuture.getFutureValue(rm.getRMInitialState(counter));
```

Comment:
When you have a big number of nodes, this operation will be very costly. Ideally, this class should as well cache its state to avoid the overhead. This can be done in a subsequent pull request

Reply:
forget about my comment, I didn't look at the code well enough. this already does the job as you use the rmproxy with a counter.